bin/integration_test.sh

```diff
@@ -0,0 +1,87 @@
+#!/bin/bash
+
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+#
+# This runs an integration test by putting crash data into the source
+# S3 bucket, invoking the submitter with an event, and then checking
+# the dest S3 bucket to see if the bits are there.
+
+set -e
+
+HOSTUSER="$(id -u):$(id -g)"
+
+SOURCEDIR="./fakedata_source/"
+SOURCEBUCKET="s3://source_bucket/"
+
+DESTDIR="./fakedata_dest/"
+DESTBUCKET="s3://dest_bucket/"
+
+# Start antenna and localstack-s3 containers
+echo ">>> SETUP"
+
+# Delete and recreate DESTDIR
+rm -rf ${DESTDIR}
+mkdir ${DESTDIR}
+
+# Start localstack-s3 and make sure buckets are clean
+docker-compose up -d localstack-s3
+# NOTE(willkg): This is localhost:5000 from the host and localstack-s3:5000
+# from inside the containers.
+./bin/wait.sh localhost 5000
+
+# FIXME(willkg): check the bucket and delete it if it exists
+docker-compose run -u "${HOSTUSER}" test bash -c "
+    ./bin/aws_s3.sh rb --force ${SOURCEBUCKET} 2>&1 > /dev/null;
+    ./bin/aws_s3.sh mb ${SOURCEBUCKET};
+"
+docker-compose run -u "${HOSTUSER}" test bash -c "
+    ./bin/aws_s3.sh rb --force ${DESTBUCKET} 2>&1 > /dev/null;
+    ./bin/aws_s3.sh mb ${DESTBUCKET}
+"
+# Start antenna
+docker-compose up -d antenna
+# NOTE(willkg): This is localhost:8888 from the host and antenna:8888
+# from inside the containers.
+./bin/wait.sh localhost 8888
+
+# Get a crash id from the fakecrashdata directory
+# CRASHID=$(find fakecrashdata/ -type f | grep raw_crash | awk -F / '{print $6}')
+CRASHID="11107bd0-2d1c-4865-af09-80bc00180313"
+CRASHKEY="v2/raw_crash/${CRASHID:0:3}/20${CRASHID:30:6}/${CRASHID}"
+
+# Copy source crash data into S3 source bucket
+docker-compose run -u "${HOSTUSER}" test ./bin/aws_s3.sh sync ${SOURCEDIR} ${SOURCEBUCKET}
+
+# Gemerate am event
+echo ">>> GENERATE AN EVENT"
+EVENT=$(./bin/generate_event.py --bucket source_bucket --key ${CRASHKEY})
+
+# Loop until one of the invokes accepts the event and submits it to the
+# destination collector
```

Comment:
This tests the function with the default throttle, so it'll run until the first one submits.

I'm not sure what you're suggesting we change that to.

Reply:
I'm suggesting we disable the throttling (by setting it to 100, so it would never bother throttling anything) during this test so that we don't have to loop until it accepts the event; we can just run once and be assured that the event will be accepted. 

This removes the need for a loop at all and simplifies the test script. I'm suggesting that it's worth doing because having the loop as part of the integration test isn't adding any useful testing to this script because we aren't actually testing that the throttle is functioning correctly, we're just working around it.

Reply:
Oh, I see. That makes sense.

I actually like watching it looping and throttling. That was helpful. We could set it up to run once with 100 throttle and once with 0 throttle. That'd work, too. Would you prefer that?

Reply:
> I actually like watching it looping and throttling.

lol, maybe just make it an optional parameter for the script then?

> We could set it up to run once with 100 throttle and once with 0 throttle.

We can assert useful things in those cases (that, at 0 throttle, nothing was accepted, and at 100 throttle it was), so yeah, that would work.